client/galaxy/scripts/mvc/dataset/dataset-edit-attributes.js

```diff
@@ -0,0 +1,250 @@
+
+define( [ 'utils/utils', 'mvc/ui/ui-tabs', 'mvc/ui/ui-misc', 'mvc/form/form-view' ], function( Utils, Tabs, Ui, Form ) {
+
+    /** Dataset edit attributes view */
+    var View = Backbone.View.extend({
+
+        initialize: function( dataset_id ) {
+            this.setElement( '<div/>' );
+            this.render( dataset_id );
+        },
+
+        // Fetch data for the selected dataset and 
+        // build tabs for editing its attributes
+        render: function( dataset_id ) {
+            var url = Galaxy.root + 'dataset/edit',
+                self = this;
+            Utils.get({
+                url     : url,
+                data    : { 'dataset_id' : dataset_id },
+                success : function( response ) {
+                   self.render_attribute_page( self, response );
+                },
+                error   : function( response ) {
+                    var error_response = { 'status': 'error',
+                        'message': 'Error occured while loading the dataset.' };
+                    self.display_message( self, error_response, self.$el.find( '.edit-attr' ) );
+                }
+            });
+        },
+
+        /** Render all the tabs view */
+        render_attribute_page: function( self, response ) {
+            var $el_edit_attr = null;
+            self.$el.empty().append( self._templateHeader() );
+            $el_edit_attr = self.$el.find( '.edit-attr' );
+            self.display_message( self, response, $el_edit_attr );
+            // Create all tabs
+            self.create_tabs( self, response, $el_edit_attr );
+        },
+
+        /** Perform AJAX post call */
+        call_ajax: function( self, data, tab_name ) {
+            var post_url = Galaxy.root + 'dataset/edit';
+            $.ajax({
+                type: "PUT",
+                url: post_url,
+                data: data,
+                success: function( response ) {
+                    self.render_attribute_page( self, response );
+                    self.reload_history();
+                },
+                error   : function( response ) {
+                    var error_response = { 'status': 'error',
+                        'message': 'Error occured while saving. Please fill all the required fields and try again.' };
+                    self.display_message( self, error_response, self.$el.find( '.edit-attr' ) );
```

Comment:
I think we should Ui.Message instead of this custom message rendering if possible.